app/modules/dataframeable.rb

```diff
@@ -0,0 +1,17 @@
+module Dataframeable
+
+  def self.create_df
+    all_three = Rating.all.select('user_id, cocktail_id, stars')
+
+    all_three_array = all_three.map do |three|
+      [three.user_id, three.cocktail_id, three.stars]
+    end
+
+    df = Pandas.DataFrame.new(data = all_three_array)
+
+    df.columns=['user_id', 'cocktail_id', 'stars']
+
+    df_pivot = df.pivot(index='user_id', columns='cocktail_id', values='stars')
+    df_pivot.fillna(0)
+  end
+end
```

Comment:
We'll need to do some refactoring here as well once we've proven that this will work in production.

Reply:
Also, we should def move that model logic over to the Rating model and we'll need to call RecommendationService from the controller anyway, so we should be able to pass that as an @ variable

Reply:
(Which we'd need to pass to this method, though, I should point out)

Reply:
Yes, agreed! Can make some helper methods. Yes, lines 4-8 can go into the Ratings model.